### PR TITLE
🚸 Add loading hint & init timeout

### PR DESCRIPTION
**Pull request by nwingt**
- Guard possible errors when init `rootStore` failed
- Added hint text in loading screen to pinpointing error for CS
- Alert user to restart the app if init `rootStore` takes longer than usual

### CODE REVIEW

File: app/i18n/translations/zh-Hant-HK.json [hunk 1]
```diff
@@ -1,4 +1,10 @@
 {
+  "closeApp": "結束 App",
+  "initializing": "初始化中",
+  "initializingFailedAlertTitle": "初始化失敗",
+  "initializingFailedAlertMessage": "請重新啟動 App",
+  "signingIn": "登錄中",
```
Comment (williamchong) [inline app/i18n/translations/zh-Hant-HK.json:6]:
```suggestion
  "signingIn": "登入中",
```

File: app/i18n/translations/zh-Hant-TW.json [hunk 2]
```diff
@@ -1,4 +1,10 @@
 {
+  "closeApp": "結束 App",
+  "initializing": "初始化中",
+  "initializingFailedAlertTitle": "初始化失敗",
+  "initializingFailedAlertMessage": "請重新啟動 App",
+  "signingIn": "登錄中",
```
Comment (williamchong) [inline app/i18n/translations/zh-Hant-TW.json:6]:
```suggestion
  "signingIn": "登入中",
```

File: app/i18n/translations/zh-Hant-TW.json [hunk 3]
```diff
@@ -1,4 +1,10 @@
 {
+  "closeApp": "結束 App",
```
Comment (williamchong) [inline app/i18n/translations/zh-Hant-TW.json:2]:
```suggestion
  "closeApp": "關閉 App",
```

File: app/i18n/translations/zh-Hant-HK.json [hunk 4]
```diff
@@ -1,4 +1,10 @@
 {
+  "closeApp": "結束 App",
```
Comment (williamchong) [inline app/i18n/translations/zh-Hant-HK.json:2]:
```suggestion
  "closeApp": "關閉 App",
```

File: .env [hunk 5]
```diff
@@ -4,6 +4,8 @@
 # Then just do this to import them:
 # import { API_URL } from 'react-native-dotenv'
 
+INIT_APP_TIMEOUT=10000
```
Comment (williamchong) [inline .env:7]: 10000 seems short, what is the average value we are expecting if there is no bug?

**Review by williamchong — CHANGES_REQUESTED** [review]

File: .env (hunk 5 above)
Comment (nwingt) [reply .env:7]: less than a sec

File: app/i18n/translations/zh-Hant-TW.json [hunk 6]
```diff
@@ -1,4 +1,10 @@
...
+  "initializing": "初始化中",
```
Comment (williamchong) [inline app/i18n/translations/zh-Hant-TW.json:3]:
```suggestion
  "initializing": "起動中",
```
`初始化中` 感覺很機械， `起動中` 會否較好？

File: .env (hunk 5 above)
Comment (williamchong) [reply .env:7]: ok stick with 10000 then

File: app/i18n/translations/zh-Hant-TW.json (hunk 6 above)
Comment (williamchong) [reply app/i18n/translations/zh-Hant-TW.json:3]: or 啓動中

File: app/i18n/translations/zh-Hant-HK.json [hunk 7]
```diff
@@ -1,4 +1,10 @@
 {
+  "closeApp": "關閉 App",
+  "initializing": "初始化中",
```
Comment (williamchong) [inline app/i18n/translations/zh-Hant-HK.json:3]:
```suggestion
  "initializing": "啟動中",
```

File: app/i18n/translations/zh-Hant-HK.json [hunk 8]
```diff
@@ -1,4 +1,10 @@
...
+  "initializingFailedAlertTitle": "初始化失敗",
```
Comment (williamchong) [inline app/i18n/translations/zh-Hant-HK.json:4]:
```suggestion
  "initializingFailedAlertTitle": "啟動失敗",
```

File: app/i18n/translations/zh-Hant-TW.json [hunk 9]
```diff
@@ -1,4 +1,10 @@
 {
+  "closeApp": "關閉 App",
+  "initializing": "起動中",
```
Comment (williamchong) [inline app/i18n/translations/zh-Hant-TW.json:3]:
```suggestion
  "initializing": "啟動中",
```

File: app/i18n/translations/zh-Hant-TW.json [hunk 10]
```diff
@@ -1,4 +1,10 @@
...
+  "initializingFailedAlertTitle": "初始化失敗",
```
Comment (williamchong) [inline app/i18n/translations/zh-Hant-TW.json:4]:
```suggestion
  "initializingFailedAlertTitle": "啟動失敗",
```

File: app/i18n/translations/zh-Hant-TW.json [hunk 11]
```diff
@@ -1,4 +1,10 @@
...
+  "initializingFailedAlertMessage": "請重新啟動 App",
```
Comment (williamchong) [inline app/i18n/translations/zh-Hant-TW.json:5]:
```suggestion
  "initializingFailedAlertMessage": "請重新開啟 App",
```

File: app/i18n/translations/zh-Hant-HK.json [hunk 12]
```diff
@@ -1,4 +1,10 @@
...
+  "initializingFailedAlertMessage": "請重新啟動 App",
```
Comment (williamchong) [inline app/i18n/translations/zh-Hant-HK.json:5]:
```suggestion
  "initializingFailedAlertMessage": "請重新開啟 App",
```

**Review by williamchong — CHANGES_REQUESTED** [review]

**Review by williamchong — APPROVED** [review]

**Pull request merged**